src/common/abstractPromote.ts

```diff
@@ -77,18 +79,25 @@ export abstract class PromoteCommand<T extends typeof SfCommand> extends SfComma
     this.sourceStageId = this.getSourceStageId();
     const asyncOperationId: string = await this.requestPromotion(doceOrg);
 
-    // TODO: move this to logger service
-    this.log(`Job ID: ${asyncOperationId}`);
-    await DeployPipelineCache.set(asyncOperationId, {});
+    const outputService: OutputService = OutputServiceFactory.forDeployment(doceOrg.getConnection());
+    outputService.printAorId(asyncOperationId);
 
     if (this.flags.async) {
-      // TODO display async message
+      await DeployPipelineCache.set(asyncOperationId, {});
+      outputService.printAsyncRunInfo(asyncOperationId);
+
       return {
         jobId: asyncOperationId,
         status: AsyncOperationStatus.InProgress,
       };
+    } else {
+      await outputService.printProgressSummary(asyncOperationId, this.flags['branch-name']);
+
+      const doceMonitor: DoceMonitor = new AsyncOpStreaming(doceOrg, this.flags.wait, asyncOperationId, outputService);
+      await doceMonitor.monitor();
```

Comment:
This falls through and on line 100 we monitor again.  Is that intentional?

Reply:
It was a merge problem, it is now fixed.